mesonbuild/environment.py

```diff
@@ -813,6 +813,11 @@ def detect_d_compiler(self, want_cross):
         # up to date language version at time (2016).
         if 'DC' in os.environ:
             exelist = shlex.split(os.environ['DC'])
+            for dc in exelist[:]:
+                if os.path.basename(dc).startswith(['ldmd', 'gdmd']):
+                    exelist.remove(dc)
```

Comment:
Sorry for not mentioning this earlier, but this should be accompanied with a log message saying that we ignored these frontends, why we ignored them, and what the user should do instead. Otherwise people will just get confused and waste time.

Reply:
Also, test failure is relevant: 

> TypeError: startswith first arg must be str or a tuple of str, not list

So it should be `startswith(('ldmd', 'gdmd'))`.